src/pages/sign/signUp/page.tsx

```diff
@@ -1,11 +1,174 @@
-import { Layout } from "antd";
+import { Form, Input, Layout, Button } from "antd";
+import uuid from "react-uuid";
+import { useUnit } from "effector-react";
 import { PageHeader } from "../../../shared/ui/PageHeader/PageHeader";
+import { $users, addUser, signupFormSubmitted } from "./model";
+import "./page.css";
 
 const { Content } = Layout;
 
-export const SignUpPage = () => (
-  <Layout>
-    <PageHeader title="Регистрация" />
-    <Content />
-  </Layout>
-);
+type User = {
+  nickName: string;
+  name: string;
+  surname: string;
+  patronymic: string;
+  password: string;
+  confirm: never;
+};
+
+const deleteConfirmAddId = (user: User) => {
+  const userData = { ...user, id: uuid() };
+  delete userData.confirm;
+  return userData;
+};
+
+export const SignUpPage = () => {
+  const [users, addUserFn, signupFormSubmittedFn] = useUnit([
+    $users,
+    addUser,
+    signupFormSubmitted,
+  ]);
+
+  const onFinish = (values: User) => {
+    addUserFn(deleteConfirmAddId(values));
+    signupFormSubmittedFn();
```

Comment:
I think that you can not call signupFormSubmittedFn in the handler, can you also transfer this to the effector model?
eg:
```
redirect({
  clock: addUserFn,
  route: routes.student,
});
```